File: lab4_approximation/src/main/java/Utils/StringFormatter.java
package Utils;

import java.util.Arrays;
import java.util.Formatter;
import java.util.IllegalFormatException;

public class StringFormatter {

    public static String center(String fmtStr, Formatter fmt, Object obj, int width) {
        String str;
        try {
            Formatter tmp = new Formatter();
            tmp.format(fmtStr, obj);
            str = tmp.toString();
        } catch(IllegalFormatException exc) {
            System.out.println("Неверный запрос формата\n");
            fmt.format("");
            return String.valueOf(fmt);
        }
        int dif = width - str.length();
        if(dif < 0) {
            fmt.format(str);
            return String.valueOf(fmt);
        }
        char[] pad = new char[dif/2];
        Arrays.fill(pad, ' ');
        fmt.format(new String(pad));
        fmt.format(str);
        pad = new char[width-dif/2-str.length()];
        Arrays.fill(pad, ' ');
        fmt.format(new String(pad));
        return String.valueOf(fmt);
    }

    public static String formatStingCenter (String line, int n) {
        return center("%s", new Formatter(), line, n);
    }
}

File: lab6_differentiation/src/main/java/StringFormatter.kt
import java.util.*

object StringFormatter {
    fun center(fmtStr: String?, fmt: Formatter, obj: Any?, width: Int): String {
        val str: String
        str = try {
            val tmp = Formatter()
            tmp.format(fmtStr, obj)
            tmp.toString()
        } catch (exc: IllegalFormatException) {
            println("Неверный запрос формата\n")
            fmt.format("")
            return fmt.toString()
        }
        val dif = width - str.length
        if (dif < 0) {
            fmt.format(str)
            return fmt.toString()
        }
        var pad = CharArray(dif / 2)
        Arrays.fill(pad, ' ')
        fmt.format(String(pad))
        fmt.format(str)
        pad = CharArray(width - dif / 2 - str.length)
        Arrays.fill(pad, ' ')
        fmt.format(String(pad))
        return fmt.toString()
    }

    fun formatStingCenter(line: String?, n: Int): String {
        return center("%s", Formatter(), line, n)
    }
}